### PR TITLE
fix: requirement test generation

### PR DESCRIPTION
**Pull request by dvarasani-crest**
Fix the issue with requirement test generation.

 - Requirement test generation for "test_cim_fields_recommended" was failing for the sample event with only `other_mappings` and no cim mapping attached.

Updated e2e tests to cover this scenario as well.

### CODE REVIEW

File: tests/e2e/addons/TA_transition_from_req/samples/sample_modinput.xml [hunk 1]
```diff
@@ -30,4 +30,23 @@
...
+        <field name="action" value="failed" />
+        <field name="status" value="OTHER" />
+        <field name="app" value="psa" />
+        <field name="src" value="1.1.1.1" />
+        <field name="user" value="admin" />
+        <field name="dest" value="so1" />
+      <field name="vendor_product" value="Pytest Splunk Addon"/>
+    </other_mappings>
+  </event>
 </device>
```
Comment (mkolasinski-splunk) [inline tests/e2e/addons/TA_transition_from_req/samples/sample_modinput.xml:52]: Please add \n at EOF

Comment (dvarasani-crest) [reply tests/e2e/addons/TA_transition_from_req/samples/sample_modinput.xml]: done

**Review by mkolasinski-splunk — APPROVED** [review]

**Review by harshilgajera-crest — APPROVED** [review]

**Pull request merged**